core/src/main/scala/cell/Cell.scala

```diff
@@ -306,8 +305,7 @@ private class CellImpl[K <: Key[V], V](pool: HandlerPool, val key: K, lattice: L
           success = true
 
         case raw: State[_, _] => // not completed
-          val newDep = new NextDepRunnable(pool, other, valueCallback, this)
-          // TODO: it looks like `newDep` is wrapped into a CallbackRunnable by `onComplete` -> bad
+          val newDep = new NextDepRunnable(pool, this, other, valueCallback)
```

Comment:
I don't understand this change. It just changes the order of arguments in `NextDepRunnable`. Why?

Reply:
This is indeed not needed, but I suggest to use the order of this PR. I would expect the two cells to be adjacent arguments.

(By the way, the *CallbackRunnable classes have a cell parameter that is not used. One could remove it.)

Reply:
OK, sounds good. However, we should also remove the unused parameter then.